var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -204,146 +309,154 @@ def url_for_version(self, version):
             update = ""
         else:
             update = "_update{0}".format(split_ver[1])
-        return "https://github.com/lammps/lammps/archive/patch_{0}{1}.tar.gz".format(
-            vdate.strftime("%d%b%Y").lstrip("0"), update
+
+        is_stable = (
+            version in self.versions
+            and "preferred" in self.versions[version]
+            and self.versions[version]["preferred"]
```

Comment:
Looking at this I understand that all stable versions should be marked `preferred` and that patch releases should never be marked preferred, is this correct?

If it is, I'm not sure to understand the comment:
> only the latest update of any given stable release should be marked preferred=True

Reply:
Stable release sometimes have update releases. If a stable has an update release, that usually means there was a serious enough bug that the stable version prior to the update shouldn't be used anymore. E.g. if there was a `.1` update release before and there is now a `.2` update release, only the `.2` should be preferred.

Reply:
@RemiLacroix-IDRIS turns out you were right, there was a logical error in that last rule, since it conflicted with the others.